packages/data/src/graphql/fetch.ts

```diff
@@ -12,19 +12,36 @@ export type GqlFetchResult<T> = {
 export const fetchGraphQl = async <T, V = Record<string, unknown>>({
   query,
   variables,
+  network,
 }: {
   query: DocumentNode | string;
   variables?: V;
+  network?: Network | '';
 }): Promise<GqlFetchResult<T>> => {
-  try {
-    const client = new GraphQLClient(GRAPHQL_ENDPOINT);
-    return {
-      data: await client.request<T>(query, variables),
-    };
-
-  } catch (error:unknown) {
-    return {
-      error: getErrorMessage(error),
-    };
+
+  const endpointReady =  mbjs?.keys?.isSet || network?.length > 0; 
+
+  let graphqlEndpoint = mbjs?.keys.graphqlUrl ?? '';
+
+  if (network && network.length > 0) {
+    graphqlEndpoint = GRAPHQL_ENDPOINTS[network];
+  }
+
+  if (endpointReady) {
```

Comment:
Overall the logic in here is a bit hard to follow.  If we don't have an endpoint configured correctly, we only need one logical operation `if(!graphQLEndpoint) { throw new Error('GraphQL endpoint not configured...'); }`. this should be unit tested as well

Reply:
so[ changed here](https://github.com/Mintbase/mintbase-js/blob/a6f94130ae6954ef6e6356f497403392c1810b16/packages/data/src/graphql/fetch.ts)


added the logic to check:

- is valid network coming from argument or mbjs.keys.network
  - if not ,return error message
- if network isnt set in any of the places
  - if not return error message
- if endpointReady = network exists and is valid, it will set graphqEndpoint
- if endpointReady && graphqlEndpoint === fetch the data